### PR TITLE
docs: clarify app database configuration with recovery

### LINKED ISSUE #4640: [Bug]: Wrong credentials in application secret after promoting a Replica Cluster
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

1.23.1

### What version of Kubernetes are you using?

1.29

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

Given a source Cluster `cluster-source` that specifies a non-default `user` and `database`:
```
metadata:
  name: cluster-source

bootstrap:
    initdb:
      database: mydb
      owner: myuser
```

And a Replica Cluster `cluster-replica`:
```
metadata:
  name: cluster-replica

bootstrap:
  recovery:
    source: cluster-source

replica:
  enabled: true
  source: cluster-source

externalClusters:
- name: cluster-source
  connectionParameters:
    host: cluster-source-rw.default.svc
    user: myuser
    dbname: mydb
  password:
    name: cluster-source-app
    key: password
```

Premise:
* The “-app” secret is not being generated for Replica Clusters, see [ShouldPgBaseBackupCreateApplicationSecret()](https://github.com/cloudnative-pg/cloudnative-pg/blob/main/api/v1/cluster_types.go#L3052).
* Replica Clusters will skip the application database configuration, see https://cloudnative-pg.io/documentation/1.23/bootstrap/#configure-the-application-database (we are just creating a physical replica of the source cluster)

After promoting `cluster-replica` by setting `replica.enabled` to `false`, now that the cluster is not in replica mode anymore the operator will create a `cluster-replica-app` secret using the default value `app` as `owner` and `database`.

The result is an application secret `cluster-replica-app` which contains an user `app` and a database `app` that do not exist in our newly promoted Cluster.

Also, the operator will unsuccessfully try to reconcile the password for the `app` user, which will continuously fail:
```
{"level":"info","ts":"2024-05-22T15:25:38Z","logger":"postgres","msg":"record","logging_pod":"cluster-replica-basicauth-1","record":{"log_time":"2024-05-22 15:25:38.639 UTC","user_name":"postgres","database_name":"postgres","process_id":"307","connection_from":"[local]","session_id":"664e0e72.133","session_line_num":"1","command_tag":"ALTER ROLE","session_start_time":"2024-05-22 15:25:38 UTC","virtual_transaction_id":"3/396","transaction_id":"0","error_severity":"ERROR","sql_state_code":"42704","message":"role \"app\" does not exist","query":"ALTER ROLE \"app\" WITH PASSWORD 'npWLS0TdGc9Ya9NQER29hlXN6dHq4wvKKeuYQXjJHjrBy8VnltJtx9FoMPnP9oaz'","application_name":"cnpg-instance-manager","backend_type":"client backend","query_id":"0"}}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mnencia**
Closes #4640

### CODE REVIEW

**Review by NiccoloFei — APPROVED** [review]

**Comment by gbartolini** [comment]
I think the proposed documentation misses the point that, until the Postgres cluster is in recovery, any existing local secret containing a password is ignored. Passwords are coming directly from the source Postgres, and only if/when the local cluster is promoted, local secrets are evaluated and passwords changed, if necessary.

**Review by jsilvela — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]

File: docs/src/bootstrap.md [hunk 1]
```diff
@@ -594,15 +594,23 @@ spec:
...
 If the new cluster is created as a replica cluster (with replica mode enabled), application
 database configuration will be skipped.
 
-The following example configure the application database `app` with password in
-supplied secret `app-secret` after bootstrap from a live cluster.
+!!! Important
+    While the `Cluster` is in recovery mode, no changes to the database,
+    including the catalog, are permitted. This restriction includes any role
+    overrides, which are deferred until the `Cluster` transitions to primary.
+    During this phase, users remain as defined in the source cluster.
```
Comment (jsilvela) [inline docs/src/bootstrap.md:609]: I'd be explicit about "this" here. Also, "roles" to be consistent with what we say further down.
```suggestion
    During the recovery phase, roles remain as defined in the source cluster.
```

File: docs/src/recovery.md [hunk 2]
```diff
@@ -87,6 +87,11 @@ spec:
           maxParallel: 8
 ```
 
+The previous example assumes that the application database and its owning user
+are named `app` by default. If the PostgreSQL cluster being restored uses
+different names upon exiting the recovery phase, you must specify these names
```
Comment (jsilvela) [inline docs/src/recovery.md:92]: IMO the comma is needed here. Also, "upon" is not preventative
```suggestion
different names, upon exiting the recovery phase, you must specify these names
```

Might even rephrase:
```suggestion
different names, you must specify these names before exiting the recovery phase,
```

File: docs/src/recovery.md [hunk 3]
```diff
@@ -193,26 +203,29 @@ spec:
...
-  from the backup that's being restored. The operator doesn't currently attempt
-  to back up the underlying secrets, as this is part of the usual maintenance
-  activity of the Kubernetes cluster.
-- To preserve the original postgres user password, you need to properly
-  configure `enableSuperuserAccess` and supply a `superuserSecret`.
-- By default, the recovery continues up to the latest
-  available WAL on the default target timeline (`latest`).
-  You can optionally specify a `recoveryTarget` to perform a point-in-time
-  recovery (see [Point in time recovery (PITR)](#point-in-time-recovery-pitr)).
+- The application database name and user are preserved from the backup being
```
Comment (jsilvela) [inline docs/src/recovery.md:220]: IMO "preserved" is a bit at odds with "restore".
```suggestion
- The application database name and user are copied from the backup being
```

File: docs/src/replica_cluster.md [hunk 4]
```diff
@@ -35,6 +35,34 @@ create and synchronize a PostgreSQL cluster from an existing source cluster
...
+A replica cluster operates in continuous recovery mode, meaning no changes to
+the database, including the catalog and global objects like roles or databases,
+are permitted. These changes are deferred until the `Cluster` transitions to
+primary. During this phase, global objects such as roles remain as defined in
+the source cluster. CloudNativePG applies any local redefinitions once the
+cluster is promoted.
+
+If you are not planning to promote the cluster (e.g., for read-only workloads)
+or if you intend to detach completely from the source of the replica cluster
+once it is promoted, you don't need to take any action.
```
Comment (jsilvela) [inline docs/src/replica_cluster.md:49]:
```suggestion
or if you intend to detach completely from the source cluster
once the replica cluster is promoted, you don't need to take any action.
```

File: docs/src/replica_cluster.md [hunk 5]
```diff
@@ -35,6 +35,34 @@ create and synchronize a PostgreSQL cluster from an existing source cluster
...
+This is normally the case of the ["Standalone Replica Cluster"](replica_cluster.md#standalone-replica-clusters).
+
+If you are planning to promote the cluster at some point, CloudNativePG will
+take care of the following roles and passwords when transitioning from replica
```
Comment (jsilvela) [inline docs/src/replica_cluster.md:53]:
```suggestion
manage the following roles and passwords when transitioning from replica
```

**Review by jsilvela — COMMENTED** [review]
A few suggestions, optional

**Review by leonardoce — APPROVED** [review]

**Pull request merged**